Create TeacherDetails, PresentationGroup and RequestTableGroupAndMentor Models

kunaladarsh opened the pull request.

kunaladarsh committed:
feat: create teacherDetails model with validation and schema definitions

kunaladarsh committed:
feat: create teacherDetails model with validation and schema definitions

kunaladarsh committed:
feat: Add PresentationGroup and RequestTableGroupAndMentor models

ShivaShukla08 commented on models/TeacherDetailModel.js:
```diff
@@ -0,0 +1,136 @@
+        require: [true, 'Please enter your Id'],
+        unique: true,
+        trim: true,
+        lowercase: true
```
Hi Adarsh, I think tid will be consisting of digits only. So there is no need of lowercase.

ShivaShukla08 commented on models/TeacherDetailModel.js:
```diff
@@ -0,0 +1,136 @@
+    },
+
+    gender: {
+        type: String,
```
Here give your own set of allowed gender categories, so that no one can enter some random string.

kunaladarsh replied:
okay

kunaladarsh committed:
update tid field

ShivaShukla08 commented on models/TeacherDetailModel.js:
```diff
@@ -0,0 +1,134 @@
+    },
+
+    personalMail: {
+        type: String,
```
Add validator

kunaladarsh committed:
refactor: add validator in email

kunaladarsh committed:
Update PresentationGroupsModel.js

Update: Disable automatic _id generation for RemovingForcefully, groupWorkRole, and removingVote subdocuments.

kunaladarsh committed:
Update RequestTableGroupAndMentorSchema and Add requestedByStudents Field

- Modified the schema for the message.reject field to prevent automatic _id generation for subdocuments.
-Add requestedByStudents field to track students who have sent requests to mentors.

kunaladarsh committed:
update teacher Deatail Model

ShivaShukla08 reviewed (approved):
Great! Work Adarsh
